Applications/Calculator/Calculator.cpp

```diff
@@ -104,18 +107,34 @@ double Calculator::finish_operation(double argument)
         return argument;
 
     case Operation::Add:
+        if ((argument > MAX - m_saved_argument) || (m_saved_argument > MAX + argument)) {
+            m_has_error = true;
+            return argument;
+        }
```

Comment:
Why not use `builtin_add_overflow()` or something?

Reply:
How about instead of an error Boolean we would use an enum like structure with a code for each possible error?

Reply:
Well it still displays as `ERR` in the UI, so what would be the point of that?

Reply:
For example, in some calculators when you divide by zero a message is displayed on the screen saying that `division by zero is not possible` or something like that. But yea, maybe this is an overkill.

Reply:
Where is `builtin_add_overflow()` defined? I'm guessing it works like the gcc extension, which seems like a good fit for this.

Reply:
It's not defined anywhere; it *is* the GCC extension :)

Reply:
I see. Implementing that works fine for everything aside from division, as there is no specific function for division.
This function `__builtin_mul_overflow_p(m_saved_argument, ((1 /argument), &res)` works logically but, as far as I can tell, you can't use that function with doubles. Any help?